WordPress/WordPressNotificationContentExtension/Sources/NotificationViewController.swift

```diff
@@ -66,19 +75,51 @@ class NotificationViewController: UIViewController {
 extension NotificationViewController: UNNotificationContentExtension {
     func didReceive(_ notification: UNNotification) {
         let notificationContent = notification.request.content
-        viewModel = RichNotificationViewModel(notificationContent: notificationContent)
+
+        guard let data = notificationContent.userInfo[CodingUserInfoKey.richNotificationViewModel.rawValue] as? Data else {
+            return
+        }
+        viewModel = RichNotificationViewModel(data: data)
 
         let username = readExtensionUsername()
         tracks.wpcomUsername = username
 
         let token = readExtensionToken()
         tracks.trackExtensionLaunched(token != nil)
+
+        markNotificationAsReadIfNeeded()
     }
 }
 
-// MARK: - Private behavior (Tracks)
+// MARK: - Private behavior
 
 private extension NotificationViewController {
+    /// Attempts to mark the notification as read via the remote service. This promotes UX consistency between the
+    /// Notification received and how it is rendered on the Notifications tab within the app.
+    ///
+    func markNotificationAsReadIfNeeded() {
+        guard let token = readExtensionToken(),
+            let viewModel = viewModel,
+            let notificationIdentifier = viewModel.notificationIdentifier,
+            let notificationHasBeenRead = viewModel.notificationReadStatus,
+            notificationHasBeenRead == false else {
+
+            return
+        }
+
+        let api = WordPressComRestApi(oAuthToken: token)
+        let service = NotificationSyncServiceRemote(wordPressComRestApi: api)
+        self.notificationService = service
```

Comment:
I'm just trying to understand the idiom... What is the benefit of creating a local variable and then assigning it to the instance variable? Wouldn't this be the same?

`notificationsService = NotificationSyncServiceRemote(wordPressComRestApi: api)`

I tend to make `self` explicit as much as possible, in my opinion, doing so removes a lot of noise from the code, so that's why I am asking, in particular considering that other times the instance variable is accessed, it's done without `self`

Reply:
In this scope, `service` is non-optional. Subsequently it can be used on line 114 without optional chaining.
The use of `self.` is intended to emphasize the difference.

I don't feel that strongly about it, so I can make this change.